Add USELESS_ELVIS_LEFT_IS_NULL diagnostic for redundant elvis with null literal

jitokim opened the pull request:
Tracking issue: [KT-79690](https://youtrack.jetbrains.com/issue/KT-79690)

cypressious commented on compiler/frontend/src/org/jetbrains/kotlin/diagnostics/Errors.java:
```diff
@@ -1243,6 +1243,8 @@ enum BadNamedArgumentsTarget {
             DiagnosticFactory1.create(WARNING, PositioningStrategies.USELESS_ELVIS);
     DiagnosticFactory0<KtBinaryExpression> USELESS_ELVIS_RIGHT_IS_NULL =
             DiagnosticFactory0.create(WARNING, PositioningStrategies.USELESS_ELVIS);
+    DiagnosticFactory0<KtBinaryExpression> USELESS_ELVIS_LEFT_IS_NULL =
```
Please don't modify K1 sources (`compiler/frontend`), it is no longer developed.

cypressious commented on compiler/fir/checkers/gen/org/jetbrains/kotlin/fir/analysis/diagnostics/FirErrors.kt:
```diff
@@ -867,6 +867,7 @@ object FirErrors : KtDiagnosticsContainer() {
     val NOT_NULL_ASSERTION_ON_CALLABLE_REFERENCE: KtDiagnosticFactory0 = KtDiagnosticFactory0("NOT_NULL_ASSERTION_ON_CALLABLE_REFERENCE", WARNING, SourceElementPositioningStrategies.OPERATOR, KtExpression::class, getRendererFactory())
     val USELESS_ELVIS: KtDiagnosticFactory1<ConeKotlinType> = KtDiagnosticFactory1("USELESS_ELVIS", WARNING, SourceElementPositioningStrategies.USELESS_ELVIS, KtBinaryExpression::class, getRendererFactory())
     val USELESS_ELVIS_RIGHT_IS_NULL: KtDiagnosticFactory0 = KtDiagnosticFactory0("USELESS_ELVIS_RIGHT_IS_NULL", WARNING, SourceElementPositioningStrategies.USELESS_ELVIS, KtBinaryExpression::class, getRendererFactory())
+    val USELESS_ELVIS_LEFT_IS_NULL: KtDiagnosticFactory0 = KtDiagnosticFactory0("USELESS_ELVIS_LEFT_IS_NULL", WARNING, SourceElementPositioningStrategies.USELESS_ELVIS, KtBinaryExpression::class, getRendererFactory())
```
This file is generated. Please modify `compiler/fir/checkers/checkers-component-generator/src/org/jetbrains/kotlin/fir/checkers/generator/diagnostics/FirDiagnosticsList.kt` and run the `Generate_FIR_Checker_Components_and_FIR_IDE_Diagnostics` configuration to generate all files.

cypressious commented on compiler/testData/diagnostics/tests/nullableTypes/uselessElvis.kt:
```diff
@@ -48,6 +48,21 @@ fun testFrom13648() {
 
 fun bar() = <!UNRESOLVED_REFERENCE!>unresolved<!>
 
+fun testUselessElvisLeftIsNull() {
```
Make sure to run `DiagnosticTestGenerated` and `PhasedJvmDiagnosticPsiTestGenerated`.

cypressious reviewed (changes requested):
Thank you for the PR. I've requested some changes.

jitokim commented:
Thanks for the review, @cypressious .
Removed K1 changes, updated the FIR generator source, and regenerated the files.
Updated the test data, and confirmed that both `DiagnosticTestGenerated` and `PhasedJvmDiagnosticPsiTestGenerated` pass locally.

cypressious commented:
Fixed some failing tests and merged as 9bf8ff0bdc86ec4fc22970c3daa1c6b10df3597f.
